python/cloudtik/core/_private/utils.py

```diff
@@ -1667,6 +1667,8 @@ def is_use_internal_ip(config: Dict[str, Any]) -> bool:
 
 
 def _is_use_internal_ip(provider_config: Dict[str, Any]) -> bool:
+    if provider_config["type"] == "kubernetes":
```

Comment:
The same here, to do special check according to the specific provider type should avoided. This is the principle for design of provider.

The default of use_internal_ips can put into the Kubernetes default.yaml file (True) other than do special check here.